modules/internal/DefaultAssociative.chpl

```diff
@@ -150,35 +150,19 @@ module DefaultAssociative {
           var end = new ioLiteral("}");
 
           while true {
-            // Try reading an end curly
-            f <~> end;
-            if f.error() == EFORMAT {
-              // didn't find a curly, OK
-              f.clearError();
-            } else {
-              // Stop reading if we got to the end
-              // or if there was another error.
+            // Try reading an end curly. If we get it, break.
+            try {
+              f <~> end;
               break;
-            }
+            } catch err: BadFormatError {}
```

Comment:
Generally speaking, a `catch { }` block is really bad style. I'd appreciate it very much if in this PR you could put a comment inside each of these to say something like "Didn't read end, so forget we tried, and keep reading".